sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreResult.java

```diff
@@ -119,15 +119,8 @@ private static void setRequestCharge(StoreResponse response, CosmosException cos
                     Double.toString(totalRequestCharge));
         }
         // Set total charge as final charge for the response.
-        else if (response.getResponseHeaderNames() != null) {
-            for (int i = 0; i < response.getResponseHeaderNames().length; ++i) {
-                if (Strings.areEqualIgnoreCase(
-                        response.getResponseHeaderNames()[i],
-                        HttpConstants.HttpHeaders.REQUEST_CHARGE)) {
-                    response.getResponseHeaderValues()[i] = Double.toString(totalRequestCharge);
-                    break;
-                }
-            }
+        else if (response.getResponseHeaders() != null) {
+            response.getResponseHeaders().put(HttpConstants.HttpHeaders.REQUEST_CHARGE, Double.toString(totalRequestCharge));
```

Comment:
Rntbd generated response headerMap is ImmutableMap.
now that the intermediate copied version is removed are we modifying the rntbd header map directly? if so this may not work.

Could you please check.